client/app/bundles/course/assessment/submission/pages/SubmissionsIndex/index.jsx

```diff
@@ -326,32 +324,146 @@ class VisibleSubmissionsIndex extends React.Component {
       <ConfirmationDialog
         open={publishConfirmation}
         onCancel={() => this.setState({ publishConfirmation: false })}
         onConfirm={() => {
-          dispatch(publishSubmissions());
+          dispatch(publishSubmissions(handlePublishParams));
           this.setState({ publishConfirmation: false });
         }}
-        message={<FormattedMessage {...translations.publishConfirmation} />}
+        message={
+          <FormattedMessage
+            {...translations.publishConfirmation}
+            values={values}
+          />
+        }
+      />
+    );
+  }
+
+  renderForceSubmitConfirmation(shownSubmissions, handleForceSubmitParams) {
+    const { dispatch, assessment } = this.props;
+    const { forceSubmitConfirmation } = this.state;
+    const values = {
+      unattempted: shownSubmissions.filter(
+        (s) => s.workflowState === workflowStates.Unstarted,
+      ).length,
+      attempting: shownSubmissions.filter(
+        (s) => s.workflowState === workflowStates.Attempting,
+      ).length,
+      selectedUsers: selectedUserType[handleForceSubmitParams],
+    };
+    const message = assessment.autograded
+      ? translations.forceSubmitConfirmationAutograded
+      : translations.forceSubmitConfirmation;
+
+    return (
+      <ConfirmationDialog
+        open={forceSubmitConfirmation}
+        onCancel={() => this.setState({ forceSubmitConfirmation: false })}
+        onConfirm={() => {
+          dispatch(forceSubmitSubmissions(handleForceSubmitParams));
+          this.setState({ forceSubmitConfirmation: false });
+        }}
+        message={<FormattedMessage {...message} values={values} />}
       />
     );
   }
 
   render() {
-    const { isLoading, notification } = this.props;
+    const { isLoading, notification, submissions } = this.props;
+    const {
+      includePhantoms,
+      tab,
+      publishConfirmation,
+      forceSubmitConfirmation,
+    } = this.state;
     if (isLoading) {
       return <LoadingIndicator />;
     }
+
+    const myStudentSubmissions = includePhantoms
+      ? submissions.filter(
+          (s) => s.courseUser.isStudent && s.courseUser.myStudent,
+        )
+      : submissions.filter(
+          (s) =>
+            s.courseUser.isStudent &&
+            s.courseUser.myStudent &&
+            !s.courseUser.phantom,
+        );
+    const studentSubmissions = includePhantoms
+      ? submissions.filter((s) => s.courseUser.isStudent)
+      : submissions.filter(
+          (s) => s.courseUser.isStudent && !s.courseUser.phantom,
+        );
+    const staffSubmissions = includePhantoms
+      ? submissions.filter((s) => !s.courseUser.isStudent)
+      : submissions.filter(
+          (s) => !s.courseUser.isStudent && !s.courseUser.phantom,
+        );
+    const filteredSubmissions = {
+      myStudentSubmissions,
+      studentSubmissions,
+      staffSubmissions,
+    };
+
+    const handleMyStudentsParams = includePhantoms
+      ? 'my_students_w_phantom'
```

Comment:
Would be good to extract out these magic strings and store them as constants somewhere.

Reply:
Noted